.github/workflows/kserve-test.yml

```diff
@@ -0,0 +1,118 @@
+name: "Kserve Smoke Test"
+
+on:
+  schedule:
+    - cron: "20 4 * * 1" # once a week
+  workflow_dispatch:
+
+  # push:
+  #   branches: [main]
+  #   paths:
+  #     - "!docs"
+  #     - "!demo"
+
+  # pull_request:
+
+env:
+  KIND_CLUSTER_NAME: "kserve-testing"
+  ISVC_NAME: "caikit-tgis-isvc"
+
+jobs:
+  smoke-test:
+    name: Kserve Smoke Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Free Disk Space
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          swap-storage: true
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.5.0
+        with:
+          cluster_name: ${{ env.KIND_CLUSTER_NAME }}
+
+      - name: Install kserve
+        run: |
+          # Use the hack/quick_install.sh script from the kserve repo
+          curl https://raw.githubusercontent.com/kserve/kserve/master/hack/quick_install.sh | bash
+
+      - name: Prepare metallb load balancer
```

Comment:
Can you clarify this step? Is it because you need an ingress?
If it is this the use case, in the past I used nginx for testing 
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/.github/actions/kind/action.yml#L34

Reply:
The `InferenceService` is accessed through istio, which has a service type `LoadBalancer`, which is why I installed this. A workaround that we can use to avoid installing metallb is the following:

```bash
kubectl patch deployments.apps -o yaml -n istio-system istio-ingressgateway -p '{"spec":{"template":{"spec":{"containers":[{"name":"istio-proxy","ports":[{"containerPort":8080,"hostPort":80},{"containerPort":8443,"hostPort":443}]}]}}}}' 
```

which basically exposes istio on the host and seems to do the job

Reply:
Ended up going for the istio patching route, updated.